dartagnan/src/main/java/com/dat3m/dartagnan/encoding/formulas/TupleFormulaManager.java

```diff
@@ -39,6 +42,33 @@ public Formula extract(TupleFormula f, int index) {
         return f.elements.get(index);
     }
 
+    public Formula extract(TupleFormula f, Iterable<Integer> indices) {
+        Formula extract = f;
+        for (int i : indices) {
+            if (extract instanceof TupleFormula tuple) {
+                extract = tuple.elements.get(i);
+            } else {
+                throw new IllegalArgumentException("Cannot extract from a non-tuple formula '%s'");
+            }
+        }
+        return extract;
+    }
```

Comment:
What is the purpose of passing an iterable if you will always return the element matching the last position in the iterable?

Reply:
I think you misread the code :). The `TupleFormula` can be nested and each index goes one nesting level deeper. Notice that both variables `extract` and `tuple` are modified in each iteration.

Reply:
Ok, now it makes more sense